### PR TITLE
Fix event logging context bug on non-existing endpoints

### PR DESCRIPTION
**Pull request by jonashendrickx**
Fixes 2 separate things:

- Endpoints that we haven't defined, shouldn't execute event logging middleware, i.e. '/trollz?api=123'
- While event logging is also capturing invalid api keys and api secrets, if that are not in the expected format, then we would be able to define the tenant context we'd have to log to. This means we're also unable to obtain the `FeaturesContext`. We have to make sure we avoid logging in this scenario.

Using endpoint filters we would alternatively require:

- For invalid api keys and locked api keys, be able to write directly using IGlobalStorage. Which may require a partial rewrite.
- Investigate where in the code we call event logging, because it appears to be happening in almost every layer. So it would cause a maintenance mess.

### CODE REVIEW

**Comment by abergs** [comment]
Otherwise I thinks this looks good!

File: src/AdminConsole/Middleware/EventLogStorageCommitMiddleware.cs [hunk 1]
```diff
@@ -2,14 +2,20 @@
...
-public class EventLogStorageCommitMiddleware
+public partial class EventLogStorageCommitMiddleware
 {
     private readonly RequestDelegate _next;
 
     public EventLogStorageCommitMiddleware(RequestDelegate next) => _next = next;
 
     public async Task InvokeAsync(HttpContext context, IEventLogger eventLogger)
     {
+        if (context.GetEndpoint() == null)
```
Comment (Tyrrrz) [inline src/AdminConsole/Middleware/EventLogStorageCommitMiddleware.cs:13]: A comment explaining why (what you wrote in the OP) would be nice

File: src/Api/Middleware/EventLogContextMiddleware.cs [hunk 2]
```diff
@@ -13,6 +13,12 @@ public class EventLogContextMiddleware
 
     public async Task InvokeAsync(HttpContext context, IEventLogContext eventLogContext, IFeatureContextProvider featureContextProvider)
     {
+        if (context.GetEndpoint() == null)
```
Comment (Tyrrrz) [inline src/Api/Middleware/EventLogContextMiddleware.cs:16]: Same here

File: src/Api/Middleware/EventLogStorageCommitMiddleware.cs [hunk 3]
```diff
@@ -20,6 +20,12 @@ public EventLogStorageCommitMiddleware(RequestDelegate next, ILogger<EventLogSto
     /// <param name="provider">Used to get the `IEventLogger` instance based on the current state of the request.</param>
     public async Task InvokeAsync(HttpContext context, IServiceProvider provider)
     {
+        if (context.GetEndpoint() == null)
```
Comment (Tyrrrz) [inline src/Api/Middleware/EventLogStorageCommitMiddleware.cs:23]: Same here

File: src/Service/Features/FeatureContextProvider.cs [hunk 4]
```diff
@@ -22,17 +22,25 @@ public class FeatureContextProvider : IFeatureContextProvider
...
+                {
+                    appId = ApiKeyUtils.GetAppId(httpContext.Request.Headers["ApiSecret"].ToString());
+                }
+                else if (httpContext.Request.RouteValues.ContainsKey("appId"))
+                {
+                    appId = httpContext.Request.RouteValues["appId"].ToString();
+                }
             }
-            if (httpContext.Request.Headers.ContainsKey("ApiSecret"))
+            catch (ArgumentException)
```
Comment (Tyrrrz) [inline src/Service/Features/FeatureContextProvider.cs:41]: Please add a comment when this exception can be thrown (i.e. why it's expected)

File: tests/Api.IntegrationTests/Middleware/AuthorizationIntegrationTests.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    private readonly HttpClient _client;
+
+    public AuthorizationIntegrationTests(ITestOutputHelper testOutput, PasswordlessApiFactory apiFactory)
+    {
+        apiFactory.TestOutput = testOutput;
+        _client = apiFactory.CreateClient();
+    }
+
+    [Fact]
+    public async Task ExistentEndpointProtectedByApiKey_Returns_Unauthorized_WhenBadApiKeyIsProvided()
```
Comment (Tyrrrz) [inline tests/Api.IntegrationTests/Middleware/AuthorizationIntegrationTests.cs:22]: To stay consistent with our naming schema, maybe something like this (same for others):

```suggestion
    public async Task I_cannot_access_an_existing_endpoint_with_an_invalid_API_key()
```

Comment (jonashendrickx) [reply tests/Api.IntegrationTests/Middleware/AuthorizationIntegrationTests.cs:22]: Seems less clear to me.

**Review by Tyrrrz — APPROVED** [review]

**Pull request merged**